pkgs/applications/networking/cluster/linkerd/update-edge.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl gnugrep gnused jq
+
+set -eu -o pipefail
+
+cd $(dirname "$0")
+
+TAG=$(curl ${GITHUB_TOKEN:+" -u \":$GITHUB_TOKEN\""} \
+    --silent https://api.github.com/repos/linkerd/linkerd2/releases | \
+    jq 'map(.tag_name)' | grep edge | sed 's/["|,| ]//g' | sort -r | head -n1)
```

Comment:
Only this part is different, right? Can we somehow dedupe the other parts of the script?

Reply:
No. There are other differences.

Comparing edge and stable. Differences are:
- API requests are different
- Package names are different (Influences building)
- Version names are different.
- Files to be updated are different (edge.nix and default.nix)

My first attempt was a single script with a channel parameter. Passing an argument at `passthru.updateScript`. I was having problems (permission error, bad paths, etc) by having nested shell script stored at /nix/store that I was not able to workaround. The solution I found was to split the script.